src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMasses.cpp

```diff
@@ -45,29 +45,99 @@ using namespace std;
 namespace OpenMS
 {
 
-  MultiplexDeltaMasses::MultiplexDeltaMasses(vector<double> ms) :
-    mass_shifts_(ms)
+  MultiplexDeltaMasses::MultiplexDeltaMasses()
   {
   }
 
-  void MultiplexDeltaMasses::addMassShift(double ms)
+  MultiplexDeltaMasses::MultiplexDeltaMasses(vector<MultiplexDeltaMasses::DeltaMass> dm) :
+    delta_masses_(dm)
   {
-    mass_shifts_.push_back(ms);
+  }
+
+  void MultiplexDeltaMasses::addDeltaMass(MultiplexDeltaMasses::DeltaMass dm)
+  {
+    delta_masses_.push_back(dm);
+  }
+
+  void MultiplexDeltaMasses::addDeltaMass(double ms, MultiplexDeltaMasses::LabelSet ls)
+  {
+    MultiplexDeltaMasses::DeltaMass dm = {ms, ls};
+    delta_masses_.push_back(dm);
+  }
+
+  void MultiplexDeltaMasses::addDeltaMass(double ms, String l)
+  {
+    MultiplexDeltaMasses::LabelSet ls;
+    ls.insert(l);
+    MultiplexDeltaMasses::DeltaMass dm = {ms, ls};
+    delta_masses_.push_back(dm);
   }
 
   std::vector<double> MultiplexDeltaMasses::getMassShifts() const
   {
-    return mass_shifts_;
+    std::vector<double> ms;
+    
+    /*for (unsigned i = 0; delta_masses_.size(); ++i)
+    {
+      ms.push_back(delta_masses_[i].mass_shift);
+    }*/
+    
+    for (std::vector<MultiplexDeltaMasses::DeltaMass>::const_iterator it = delta_masses_.begin(); it != delta_masses_.end(); ++it)
+    {
+      ms.push_back((*it).mass_shift);
+    }
+    
+    return ms;
+  }
+  
+  std::vector<MultiplexDeltaMasses::DeltaMass> MultiplexDeltaMasses::getDeltaMasses() const
+  {
+    return delta_masses_;
+  }
+
+  unsigned MultiplexDeltaMasses::getDeltaMassesCount() const
+  {
+    return delta_masses_.size();
   }
 
-  unsigned MultiplexDeltaMasses::getMassShiftCount() const
+  MultiplexDeltaMasses::DeltaMass MultiplexDeltaMasses::getDeltaMassAt(int i) const
   {
-    return mass_shifts_.size();
+    return delta_masses_[i];
   }
 
   double MultiplexDeltaMasses::getMassShiftAt(int i) const
   {
-    return mass_shifts_[i];
+    return delta_masses_[i].mass_shift;
   }
+  
+  MultiplexDeltaMasses::LabelSet MultiplexDeltaMasses::getLabelSetAt(int i) const
+  {
+    return delta_masses_[i].label_set;
+  }
+  
+  bool operator<(const MultiplexDeltaMasses &dm1, const MultiplexDeltaMasses &dm2)
```

Comment:
This operator doesn't impose a strict ordering and will break the use with containers that rely on.